spec/unit/command_hierarchy_spec.rb

```diff
@@ -0,0 +1,57 @@
+require '3scale_toolbox/cli'
+
+# `expect(...).not_to matcher.and matcher` is not supported
+RSpec::Matchers.define_negated_matcher :not_raise_error, :raise_error
+
+RSpec.describe 'Plugin Command Hierarchy' do
+  def create_command(command_name)
+    Class.new(Cri::CommandRunner) do
+      extend ThreeScaleToolbox::Command
+
+      this_class = self
+      define_singleton_method :command do
+        Cri::Command.define do
+          name        command_name
+          usage       command_name
+          runner      this_class
+        end
+      end
+
+      define_method :run do
+        puts "command #{command_name}"
+      end
+    end
+  end
+
+  it 'sibling commands loaded with add command' do
+    10.times.each do |cmd_idx|
+      cmd = create_command("cmd_#{cmd_idx}")
+      ThreeScaleToolbox::CLI.add_command(cmd)
+    end
+
+    10.times.each do |cmd_idx|
+      expect do
+        ThreeScaleToolbox::CLI.run(["cmd_#{cmd_idx}"])
+      end.to not_raise_error.and output("command cmd_#{cmd_idx}\n").to_stdout
+    end
+  end
+
+  it 'vertical tree loaded with add_subcommand' do
```

Comment:
This is seriously unreadable. Honestly I'd not even bother trying to understand this kind of test.

It overrides variables (`cmd`), defines dynamically content that can be handrolled, does assertion on dynamic data. All of those are test smells in my book.